Sources/InfluxDBSwift/WriteAPI.swift

```diff
@@ -300,69 +300,78 @@ public class WriteAPI {
         }
     }
 
-    private func toLineProtocol(precision: InfluxDBClient.WritePrecision,
+    private func toLineProtocol(precision: InfluxDBClient.TimestampPrecision,
                                 record: Any,
                                 defaultTags: [String: String?]?,
-                                batches: inout [InfluxDBClient.WritePrecision: (Int, [String])]) throws {
-        switch record {
-        case let string as String:
-            guard !string.trimmingCharacters(in: .whitespaces).isEmpty else {
-                return
-            }
-            if var batch = batches[precision] {
-                batch.1.append(string)
-                batches[precision] = batch
-            } else {
-                batches[precision] = (batches.count, [string])
+                                batches: inout [InfluxDBClient.TimestampPrecision: (Int, [String])]) throws {
```

Comment:
Rather than use `Any` for the `record` argument, you can overload this func:

```swift
private func toLineProtocol(precision: InfluxDBClient.TimestampPrecision,
                                record: InfluxDBClient.Point,
                                defaultTags: [String: String?]?,
                                batches: inout [InfluxDBClient.TimestampPrecision: (Int, [String])]) throws {
```

and

```swift
private func toLineProtocol(precision: InfluxDBClient.TimestampPrecision,
                                record: String,
                                defaultTags: [String: String?]?,
                                batches: inout [InfluxDBClient.TimestampPrecision: (Int, [String])]) throws {
```

etc